python/benchmarks/parse_coco.py

```diff
@@ -65,9 +65,71 @@ def read_split(split):
             return self._instances_to_df(split, json_data)
 
         df = pd.concat([read_split(split) for split in ["train", "val"]])
-        # df['date_captured'] = pd.to_datetime(df.date_captured)  # lance GH#98
+        df['date_captured'] = pd.to_datetime(df.date_captured)  # lance GH#98
         return df
 
+    def _convert_metadata_df(self, df: pd.DataFrame) -> pa.Table:
+        """Convert each metdata column to pyarrow with lance types"""
+        schema = self.get_schema()
+        arrays = []
+        for name, col in df.items():
+            field = schema.field(name)
+            arr = self._convert_field(field.name, field.type, col)
+            arrays.append(arr)
+        table = pa.Table.from_arrays(arrays, schema=schema)
+        return table
+
+    def _convert_field(self, name, typ, col):
+        if isinstance(typ, pa.ExtensionType):
+            storage = pa.array(col, type=typ.storage_type)
+            arr = pa.ExtensionArray.from_storage(typ, storage)
+        elif pa.types.is_list(typ):
+            native_arr = pa.array(col)
+            offsets = native_arr.offsets
+            values = native_arr.values.to_numpy(zero_copy_only=False)
+            return pa.ListArray.from_arrays(
+                offsets, self._convert_field(
+                    f'{name}.elements', typ.value_type, values)
+            )
+        elif pa.types.is_struct(typ):
+            native_arr = pa.array(col)
+            arrays = []
+            for subfield in typ:
+                sub_arr = native_arr.field(subfield.name)
+                if name == 'annotations' and subfield.name == 'name':
+                    converted = self._convert_name_column(
+                        sub_arr, native_arr.field('category_id')
+                    )
+                else:
+                    converted = self._convert_field(
+                        f"{name}.{subfield.name}",
+                        subfield.type,
+                        sub_arr.to_numpy(zero_copy_only=False)
+                    )
+                arrays.append(converted)
+            return pa.StructArray.from_arrays(arrays, fields=typ)
+        else:
+            arr = pa.array(col, type=typ)
+        return arr
+
+    def _convert_name_column(self, name_arr, category_id_arr):
+        coco_classes = pd.read_csv("coco_classes.csv", header=0, index_col=None)
```

Comment:
Provide this filename as parameter?

Reply:
as discussed over slack, keeping this hard coded for now to make it safer because it needs to align with category_id. We'll make this configurable when we turn this into ingestion code